### ISSUE TITLE
DatePicker browser autocomplete support

### ISSUE THREAD

**Issue by mauron85** [adobe/react-spectrum#7167]
### Provide a general summary of the feature here

Currently DatePicker from react-aria-components seems not to be supporting browser autocomplete.

I believe it's because it's implemeted as bunch of divs with contenteditable=true.

### 🤔 Expected Behavior?

DatePicker supports autocomplete functionality of the browsers

### 😯 Current Behavior

DatePicker doesn't support autocomplete.

### 🔦 Context

As I user, I often use autofill for filling up various form data. Currently it works, with Inputs, but not with DatePicker.

**Comment by LFDanLu** [comment]
We could try using a visually hidden input strategy similar to what we do with Select, where the autofill targets that input and allows us to then parse it into a date time format that is consumable by the DatePicker. Will need testing, open to any help testing it out as well!

**Comment by hades200082** [comment]
When using this component to collect date of birth (a very common use-case for this kind of component), this does not comply with WCAG Level AA 2.2:

1.3.5 Identify Input Purpose - https://www.w3.org/WAI/WCAG22/Understanding/identify-input-purpose.html

Given that the main USP of react-aria is accessibility by default, is there a timeframe for this being resolved?

**Comment by yihuiliao** [comment]
@hades200082 Sorry, just saw this! I got caught up in a different ticket so it took awhile to get to this one, but I've finally been able to dedicate more time on this issue so it should be resolved sooner or later. The PR is not currently up yet but hopefully will be soon and I'd be happy to get any feedback. 

Out of curiosity, do you use any third-party extensions to autofill forms or do you rely on the browser for that? So far, I've only been able to successfully autofill native date inputs in Safari and iOS but was wondering if you had any experience with others

**Comment by hades200082** [comment]
Our current use cases are mainly for capturing date of birth and allowing the browsers native autocomplete. 

https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/autocomplete#bday

**Comment by mauron85** [comment]
> [@hades200082](https://github.com/hades200082) Sorry, just saw this! I got caught up in a different ticket so it took awhile to get to this one, but I've finally been able to dedicate more time on this issue so it should be resolved sooner or later. The PR is not currently up yet but hopefully will be soon and I'd be happy to get any feedback.
> 
> Out of curiosity, do you use any third-party extensions to autofill forms or do you rely on the browser for that? So far, I've only been able to successfully autofill native date inputs in Safari and iOS but was wondering if you had any experience with others

@hades200082 sorry for late response. i'm using just native browser autofill.
